integrated_channels/canvas/client.py

```diff
@@ -55,43 +57,119 @@ def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-
         pass
 
     def create_content_metadata(self, serialized_data):
-        url = '{}/api/v1/accounts/{}/courses'.format(
+        url = '{}/api/v1/accounts/{}/courses/'.format(
             self.enterprise_configuration.canvas_base_url,
             self.enterprise_configuration.canvas_account_id,
         )
         self._post(url, serialized_data)
 
     def update_content_metadata(self, serialized_data):
-        # Cannot update yet since we don't have course id
-        pass
+        self._put(serialized_data)
```

Comment:
what's the point of making this a separate function? I know `update_content_metadata` is one we are overwriting from the base class, but why overwrite with another defined method rather than just stick the logic here?

1. We aren't using `_put` outside of this one context anyways (same with delete and post)
2. The incredibly generic name makes it seem like it _could_ be usable for learner data but it can't because it does a bunch of course specific logic.

I'd recommend moving the logic to these main classes and maybe use put/post/delete to pass in the specific url and just have it do requests.post/put/delete ? Or just not have those quite frankly?

Reply:
also this goes for _delete and _post as well

Reply:
I agree that `_put`, `_post` and `_delete` have grown out of their generic context. I'm going to revert all three to accepting a url and data and returning the response of their respective session calls. All other logic will go in the `update_content_metadata`, `delete_content_metadata` and `create_content_metadata`.